a3fe/tests/test_engine_configuration.py

```diff
@@ -1,25 +1,23 @@
-"""Unit and regression tests for the SomdConfig class."""
+"""Unit and regression tests for the engine configuration class."""
 
 from tempfile import TemporaryDirectory
 import os
 import pytest
 
-from a3fe import SomdConfig
 
-
-def test_config_yaml_save_and_load():
+def test_config_yaml_save_and_load(engine_config):
     """Test that the config can be saved to and loaded from YAML."""
     with TemporaryDirectory() as dirname:
-        config = SomdConfig(runtime=1)
+        config = engine_config(runtime=1)
         config.dump(dirname)
-        config2 = SomdConfig.load(dirname)
+        config2 = engine_config.load(dirname)
         assert config.runtime == config2.runtime
 
 
-def test_write_config():
-    """Test that the SOMD configuration file is generated correctly."""
+def test_write_config(engine_config):
```

Comment:
The writing config tests are probably specific to the engine, but you've (very nicely) set up `engine_config` so that it can be SOMD or GROMACS. Maybe directly supply a SOMD config here, and make the test `test_write_config_somd`? Feel free to keep it generic if you can though!